src/Symfony/Component/HttpKernel/EventListener/HttpExceptionListener.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\EventListener;
+
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Event\GetResponseForExceptionEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\HttpKernel\Exception\HttpExceptionInterface;
+
+/**
+ * HttpExceptionListener.
+ *
+ * @author Chris McCafferty <cilefen@gmail.com>
+ */
+class HttpExceptionListener implements EventSubscriberInterface
+{
+    protected $container;
+    protected $debug;
+
+    public function __construct(ContainerInterface $container, $debug = false)
+    {
+        $this->container = $container;
+        $this->debug = $debug;
+    }
+
+    public static function getSubscribedEvents()
+    {
+        return array(
+          KernelEvents::EXCEPTION => array(
+            array('onKernelException', -256),
+          ),
+        );
+    }
+
+    public function onKernelException(GetResponseForExceptionEvent $event)
+    {
+        if ($this->container->has('twig')) {
```

Comment:
instead we could remove the service definition within a compiler pass? So it would have zero runtime overhead if `twig` is available

Reply:
@dmaicher Thank you for reviewing. Yes, that's a good idea. Do you feel TwigBundle or HttpKernel should have that compiler pass logic? I think TwigBundle but I'd like to hear your opinion.

Reply:
I would create the compiler pass inside `HttpKernel` and use it accordingly in the `FrameworkBundle` like done for example here: https://github.com/symfony/symfony/blob/master/src/Symfony/Bundle/FrameworkBundle/FrameworkBundle.php#L31